br: add use constraint with TiCDC and Drainer (#3839)
pingcap/docs: br/backup-and-restore-faq.md

## What should I do to handle the `rpc error: code = Unavailable desc =...` error occurred in BR?

This error might occur when the capacity of the cluster to restore (using BR) is insufficient. You can further confirm the cause by checking the monitoring metrics of this cluster or the TiKV log.

To handle this issue, you can try to scale out the cluster resources, reduce the concurrency during restore, and enable the `RATE_LIMIT` option.

## Where are the backed up files stored when I use `local` storage?

When you use `local` storage, `backupmeta` is generated on the node where BR is running, and backup files are generated on the Leader nodes of each Region.

## How about the size of the backup data? Are there replicas of the backup?

During data backup, backup files are generated on the Leader nodes of each Region. The size of the backup is equal to the data size, with no redundant replicas. Therefore, the total data size is approximately the total number of TiKV data divided by the number of replicas.

However, if you want to restore data from local storage, the number of replicas is equal to that of the TiKV nodes, because each TiKV must have access to all backup files.

> ## What should I pay attention to when BR restores data to the upstream cluster of TiCDC/Drainer?

Reviewer: ```suggestion
## What should I do when BR restores data to the upstream cluster of TiCDC/Drainer?
```

> + **The data restored by BR cannot be replicated to the downstream**. Because BR directly imports SST files, and the downstream cluster currently cannot obtain upstream SST files.

Reviewer: ```suggestion
+ **The data restored using BR cannot be replicated to the downstream**. This is because BR directly imports SST files but the downstream cluster currently cannot obtain these files from the upstream.
```

> + Data from BR before v4.0.3 might cause unexcepted DDL executions; therefore, if you need to perform restore on the upstream cluster of TiCDC/Drainer, add all tables restored using BR to the TiCDC/Drainer block list.

Reviewer: ```suggestion
+ Data from BR before v4.0.3 might cause unexcepted DDL executions. Therefore, if you need to perform restore on the upstream cluster of TiCDC/Drainer, add all tables restored using BR to the TiCDC/Drainer block list.
```

> + Restored data that cannot be replicated to downstream might cause TiCDC/Drainer to abnormally execute DDL; therefore, if you must perform restore on the upstream cluster of TiCDC/Drainer, add all tables restored by BR to the TiCDC/Drainer block list.

Reviewer: @YuJuncen PTAL, thanks!
```suggestion
+ Restored data that cannot be replicated to the downstream might cause TiCDC/Drainer errors when DDL statements are executed. Therefore, if you need to perform restore on the upstream cluster of TiCDC/Drainer, add all tables restored using BR to the TiCDC/Drainer block list.
```

Reviewer 2: The downstream may execute unexcepted DDLs because BR didn't set the `query` field of DDL jobs properly. But for TiCDC and Binlog themselves, they probably won't notice this. So, maybe "execute unexcepted DDLs" might be a good expression.

BTW, this bug is fixed for now (after https://github.com/pingcap/br/pull/406). However, the restored data still cannot be replicated to the downstream.

How about this?
```suggestion
+ Data from older versions of BR even might cause unexcepted DDLs executed; therefore, if you must perform restore on the upstream cluster of TiCDC/Drainer, add all tables restored by BR to the TiCDC/Drainer block list.
```

Reviewer: @YuJuncen Thanks very much for explaining! I still have questions:
1. `older versions of BR` means what versions? How old the BR versions are?
2. Is the Chinese `无法被同步到下游的恢复数据可能导致 TiCDC / Drainer 在执行 DDL 的时候发生异常` a bit outdated?

Reviewer 2: 1. According to the PR, "older versions of BR" is `< v4.0.3`.
2. Yep :(

Reviewer: @YuJuncen PTAL. Also, would you create a PR to update the Chinese doc?
```suggestion
+ Data from BR before v4.0.3 might cause unexcepted DDL executions; therefore, if you need to perform restore on the upstream cluster of TiCDC/Drainer, add all tables restored using BR to the TiCDC/Drainer block list.
```

Reviewer 2: LGTM

Reviewer: Chinese updated in https://github.com/pingcap/docs-cn/pull/4454.

> You can use [`filter.rules`](https://github.com/pingcap/ticdc/blob/7c3c2336f98153326912f3cf6ea2fbb7bcc4a20c/cmd/changefeed.toml#L16) item to complete the configuration of TiCDC and use [`syncer.ignore-table`](/tidb-binlog/tidb-binlog-configuration-file.md#ignore-table) item to complete the configuration of Drainer.

Reviewer: ```suggestion
You can use [`filter.rules`](https://github.com/pingcap/ticdc/blob/7c3c2336f98153326912f3cf6ea2fbb7bcc4a20c/cmd/changefeed.toml#L16) to configure the block list for TiCDC and use [`syncer.ignore-table`](/tidb-binlog/tidb-binlog-configuration-file.md#ignore-table) to configure the block list for Drainer.
```
